Makefile

```diff
@@ -0,0 +1,42 @@
+.DEFAULT_GOAL := help
+
+PHPNOGC=php -d zend.enable_gc=0
+
+.PHONY: help
+help:
+	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
+
+
+##
+## Commands
+##---------------------------------------------------------------------------
+
+clean: 	 ## Clean all created artifacts
+.PHONY: clean
+clean:
+	git clean --exclude=.idea/ -ffdx
+
+
+build: ## Compile the application into the PHAR
+.PHONY: build
+build: bin/psysh.phar
+
+
+#
+# Rules from files
+#---------------------------------------------------------------------------
+
+composer.lock: composer.json
+	composer install
```

Comment:
if the lock file is outdated, this doesn't work. it needs to run `composer update`.

Reply:
I would eventually add an `@echo` but I wouldn't do `composer update`, updating your deps (for an app, not a lib) should be done more careful reviewing which deps needs to be updated, what changed and see that everything is compatible. Unless you have enough e2e tests to test the PHAR and being able to blindly update everything in absolute confidence